addons/mrp/models/mrp_workcenter.py

```diff
@@ -204,6 +207,45 @@ def _get_unavailability_intervals(self, start_datetime, end_datetime):
         unavailability_ressources = self.resource_id._get_unavailable_intervals(start_datetime, end_datetime)
         return {wc.id: unavailability_ressources.get(wc.resource_id.id, []) for wc in self}
 
+    def _get_first_available_slot(self, start_datetime, hour_duration):
+        """Get the first available interval for the workcenter in `self`.
+
+        The available interval is disjoinct with all other workorders planned on this workcenter, but
+        can overlap the time-off of the related calendar (inverse of the working hours).
+        Return the first available interval (start datetime, end datetime) or,
+        if there is none before 700 days, a tuple error (False, 'error message').
+
+        :param start_datetime: begin the search at this datetime
+        :param hour_duration: hour needed to make the workorder (float)
+        :rtype: tuple
+        """
+        self.ensure_one()
+        start_datetime, revert = make_aware(start_datetime)
+
+        get_intervals_work = partial(self.resource_calendar_id._work_intervals, domain=[('time_type', 'in', ['other', 'leave'])], resource=self.resource_id)
+        get_intervals_leaves_wo = partial(self.resource_calendar_id._leave_intervals, domain=[('time_type', '=', 'other')], resource=self.resource_id)
+
+        remain_duration = hour_duration
+        delta = timedelta(days=14)
+        for n in range(50):  # 50 * 14 = 700 days in advance (hardcoded)
+            dt = start_datetime + delta * n
+            intervals_leaves_wo = get_intervals_leaves_wo(dt, dt + delta)
+            intervals_work = get_intervals_work(dt, dt + delta)
+            for start, stop, meta in intervals_work:
+                interval_hours = (stop - start).total_seconds() / 3600
+                # If the remain hours has never decrease update start_datetime
+                if remain_duration == hour_duration:
+                    start_datetime = start
+                if remain_duration <= interval_hours:
+                    return revert(start_datetime), revert(start + timedelta(hours=remain_duration))
+                # If there is a intersection we doesn't validate the split in different slots
+                if (Intervals([(start, start + timedelta(hours=remain_duration), meta)]) & intervals_leaves_wo):
```

Comment:
You need 1hour of production, it's friday 4:30pm and you stop working at 5pm. A production is already booked on monday first working hour. This condition will check if there is a 'other' leave between 4:30pm and 5:30pm. It's not the case so the production will be split while it shouldn't.

Reply:
Right,  T h a n k  y o u!